devtools/zenodo/zenodo_data_release.py

```diff
@@ -0,0 +1,367 @@
+
+        Finds the latest record in the concept that record_id points to, and
+        makes a new version unless one exists already.
+        """
+        response = requests.post(
+            f"{self.base_url}/records/{record_id}/versions",
+            headers=self.auth_headers,
+            timeout=10,
+        )
+        return _NewRecord(**response.json())
+
+    def update_deposition_metadata(
+        self, deposition_id: int, metadata: _LegacyMetadata
+    ) -> _LegacyDeposition:
+        """LEGACY API: Update deposition metadata.
+
+        Replaces the existing metadata completely - so make sure to pass in
+        complete metadata. You cannot update metadata fields one at a time.
+        """
+        url = f"{self.base_url}/deposit/depositions/{deposition_id}"
+        data = {"metadata": metadata.model_dump()}
+        logger.debug(f"Setting metadata for {deposition_id} to {data}")
+        response = requests.put(url, json=data, headers=self.auth_headers, timeout=10)
+        return _LegacyDeposition(**response.json())
+
+    def delete_deposition_file(self, deposition_id: int, file_id) -> requests.Response:
+        """LEGACY API: Delete file from deposition.
+
+        Note: file_id is not always the file name.
+        """
+        return requests.delete(
+            f"{self.base_url}/deposit/depositions/{deposition_id}/files/{file_id}",
+            headers=self.auth_headers,
+            timeout=10,
+        )
+
+    def create_bucket_file(
+        self, bucket_url: str, file_name: str, file_content: BinaryIO
+    ) -> requests.Response:
+        """LEGACY API: Upload a file to a deposition's file bucket.
+
+        Prefer this over the /deposit/depositions/{id}/files endpoint because
+        it allows for files >100MB.
+        """
+        url = f"{bucket_url}/{file_name}"
+        logger.info(f"Uploading file to {url}")
+        response = requests.put(
+            url,
+            headers=self.auth_headers,
+            data=file_content,
+            timeout=3_600,  # 1 hour
+        )
+        return response
+
+    def publish_deposition(self, deposition_id: int) -> _LegacyDeposition:
+        """LEGACY API: publish deposition."""
+        response = requests.post(
+            f"{self.base_url}/deposit/depositions/{deposition_id}/actions/publish",
+            headers=self.auth_headers,
+            timeout=10,
+        )
+        return _LegacyDeposition(**response.json())
+
+
+class SourceData:
+    """Interface for retrieving the data we actually want to store."""
+
+    def iter_files(self) -> Iterable[tuple[str, BinaryIO]]:
+        """Get the names and contents of each file in the SourceData."""
+        raise NotImplementedError
+
+
+@dataclass
+class LocalSource(SourceData):
+    """Get files from local directory."""
+
+    path: Path
+
+    def iter_files(self) -> Iterable[tuple[str, BinaryIO]]:
+        """Loop through all files in the directory."""
+        for f in self.path.iterdir():
```

Comment:
It's much easier to visually inspect a record if the files are sorted, so related FERC files are next to each other.

```suggestion
        for f in sorted(self.path.iterdir()):
```